cql3/selection/abstract_function_selector.hh

```diff
@@ -66,7 +67,8 @@ public:
 
     abstract_function_selector(shared_ptr<functions::function> fun, std::vector<shared_ptr<selector>> arg_selectors)
             : _fun(std::move(fun)), _arg_selectors(std::move(arg_selectors)),
-              _requires_thread(boost::algorithm::any_of(_arg_selectors, [] (auto& s) { return s->requires_thread(); })) {
+              _requires_thread(boost::algorithm::any_of(_arg_selectors, [] (auto& s) { return s->requires_thread(); })
+                    || bool(dynamic_pointer_cast<cql3::functions::user_aggregate>(_fun))) {
```

Comment:
Better to ask the function if it requires a thread than to assume it. I see abstract_function has a requires_thread() function.

Reply:
Oh, right, it actually looks like a preexisting bug... the function itself should be asked anyway.